### PR TITLE
 bump acorn-import-assertions@1.7.3 and parse import assertions again

### LINKED ISSUE #13814: Any file js, that has an "assert" string in it, throws an error.
# Bug report

**What is the current behavior?**

After i removed the `package-lock.json` and reinstalled i got this error.

**If the current behavior is a bug, please provide the steps to reproduce.**

It is a big project, i can't reproduce, but could you please look at the error, because it only happens in files that have `assert` functions or imports.

**What is the expected behavior?**
I think it is a patch in some of the packages, so no idea where it happend.
What happens now is that any file that has `assert` in the file it will throw an error.

What is weird is that in the error it happens that has `assert` files:
```txt
[2] ERROR in ./node_modules/@uirouter/core/lib-esm/transition/hookBuilder.js 1:17
[2] Module parse failed: Unexpected token (1:17)
[2] You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
[2] > import { extend, assertPredicate, unnestR, identity } from '../common/common';
[2] | import { isArray } from '../common/predicates';
[2] | import { TransitionHookPhase, TransitionHookScope, } from './interface';
[2]  @ ./node_modules/@uirouter/core/lib-esm/transition/index.js 14:0-30 14:0-30
[2]  @ ./node_modules/@uirouter/core/lib-esm/index.js 6:0-35 6:0-35
[2]  @ ./node_modules/@uirouter/angularjs/lib-esm/index.js 16:0-39 17:0-16 18:0-31 18:0-31
[2]  @ ./src/client/vendor.js 34:0-30
[2] 
[2] ERROR in ./node_modules/@uirouter/core/lib-esm/vanilla/injector.js 1:17
[2] Module parse failed: Unexpected token (1:17)
[2] You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
[2] > import { extend, assertPredicate, isFunction, isArray, isInjectable, } from '../common/index';
[2] | // globally available injectables
[2] | var globals = {};
[2]  @ ./node_modules/@uirouter/core/lib-esm/vanilla/index.js 8:0-27 8:0-27
[2]  @ ./node_modules/@uirouter/core/lib-esm/vanilla.js 1:0-32 1:0-32
[2]  @ ./node_modules/@uirouter/core/lib-esm/index.js 11:0-26 11:0-26
[2]  @ ./node_modules/@uirouter/angularjs/lib-esm/index.js 16:0-39 17:0-16 18:0-31 18:0-31
[2]  @ ./src/client/vendor.js 34:0-30
[2] 
[2] ERROR in ./node_modules/angular-animate/angular-animate.js 64:9
[2] Module parse failed: Unexpected token (64:9)
[2] You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
[2] | 
[2] | var ngMinErr = angular.$$minErr('ng');
[2] > function assertArg(arg, name, reason) {
[2] |   if (!arg) {
[2] |     throw ngMinErr('areq', 'Argument \'{0}\' is {1}', (name || '?'), (reason || 'required'));
[2]  @ ./node_modules/angular-animate/index.js 1:0-28
[2]  @ ./node_modules/angular-material/index.js 5:0-26
[2]  @ ./src/client/vendor.js 36:0-27
[2] 
[2] ERROR in ./node_modules/angular/angular.js 2115:9
[2] Module parse failed: Unexpected token (2115:9)
[2] You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
[2] |  * throw error if the argument is falsy.
[2] |  */
[2] > function assertArg(arg, name, reason) {
[2] |   if (!arg) {
[2] |     throw ngMinErr('areq', 'Argument \'{0}\' is {1}', (name || '?'), (reason || 'required'));
[2]  @ ./node_modules/angular/index.js 1:0-20
[2]  @ ./src/client/vendor.js 24:17-35
[2] 
[2] ERROR in ./node_modules/buffer/index.js 184:9
[2] Module parse failed: Unexpected token (184:9)
[2] You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
[2] | Object.setPrototypeOf(Buffer, Uint8Array)
[2] | 
[2] > function assertSize (size) {
[2] |   if (typeof size !== 'number') {
[2] |     throw new TypeError('"size" argument must be of type number')
[2]  @ ./node_modules/qrcode/lib/core/reed-solomon-encoder.js 3:13-37
[2]  @ ./node_modules/qrcode/lib/core/qrcode.js 10:25-58
[2]  @ ./node_modules/qrcode/lib/browser.js 4:13-37
[2]  @ ./src/client/angular/core/factory/ngivr-core-pdf-generator.js 57:43-60
[2]  @ ./src/client/injector.js 80:0-61
[2]  @ ./src/client/angular.js 17:0-24
[2]  @ ./src/client/main.js 148:0-20
[2]  @ ./src/client/main-development.js 6:19-47 29:0-17
[2] 
[2] ERROR in ./node_modules/jquery/dist/jquery.js 936:9
[2] Module parse failed: Unexpected token (936:9)
[2] You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
[2] |  * @param {Function} fn Passed the created element and returns a boolean result
[2] |  */
[2] > function assert( fn ) {
[2] |   var el = document.createElement( "fieldset" );
[2] | 
[2]  @ ./src/client/vendor.js 10:27-44
[2] 
[2] webpack 5.45.0 compiled with 6 errors in 287 ms
[2] ℹ ｢wdm｣: Failed to compile.
```

**Other relevant information:**
webpack version:  5 latest
Node.js version: 14 latest
Operating System: Linux Mint latest
Additional tools: 

# See this is the main errors, which is very weird:

The actual errors all `assert`:
```txt
function assert( fn ) {
function assertSize (size) {
function assertArg(arg, name, reason) {
function assertArg(arg, name, reason) {
import { extend, assertPredicate, isFunction, isArray, isInjectable, } from '../common/index';
import { extend, assertPredicate, unnestR, identity } from '../common/common';
```

### PR DESCRIPTION
**Pull request by xtuc**
**What kind of change does this PR introduce?**

**Did you add tests for your changes?**

**Does this PR introduce a breaking change?**

**What needs to be documented once your changes are merged?**

Closes #13814

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "@webassemblyjs/wasm-edit": "1.11.1",
     "@webassemblyjs/wasm-parser": "1.11.1",
     "acorn": "^8.4.1",
-    "acorn-import-assertions": "1.7.1",
+    "acorn-import-assertions": "1.7.2",
```
Comment (alexander-akait) [inline package.json:14]: Maybe in future better to use `^` to allow developers update locally without patch release?

Comment (xtuc) [reply package.json:14]: In the past we used to pin the dependencies, but happy to change

Comment (alexander-akait) [reply package.json:14]: let's look what @sokra answer, maybe it is not bad to have parser pinned, sometimes regressions happens

Comment (sokra) [reply package.json:14]: ^

**Review by sokra — CHANGES_REQUESTED** [review]

**Review by sokra — CHANGES_REQUESTED** [review]
CI is broken. Also add a test case so we don't regress on that again

**Comment by sokra** [comment]
I wonder why this didn't fail the CI before...

**Comment by alexander-akait** [comment]
Maybe CI cache?

**Comment by xtuc** [comment]
CI failures has been introduced by acorn-import-assertions@1.7.2

**Comment by alexander-akait** [comment]
seems not all cases was fixed, `"assert"` still failed

**Comment by xtuc** [comment]
Yeah i'm fixing that right now.

File: yarn.lock [hunk 2]
```diff
@@ -1677,6 +1679,11 @@ character-parser@^2.2.0:
   dependencies:
     is-regex "^1.0.3"
 
+charcodes@^0.2.0:
```
Comment (xtuc) [inline yarn.lock:1682]: Introduced that dependency but goal is to remove https://github.com/xtuc/acorn-import-assertions/issues/9.

**Comment by sokra** [comment]
I reverted the original PR for now to fix the problem right now...

We can readd it in this PR.

**Comment by xtuc** [comment]
Sure, let me add it here again.

File: yarn.lock (hunk 2 above)
Comment (sokra) [reply yarn.lock:1682]:
``` js
import { leftCurlyBrace, space } from "charcodes";
```

can be replaced with

``` js
const leftCurlyBrace = "{".charCodeAt(0);
const space = " ".charCodeAt(0);
```

No need for a package or babel transfrom...

**Review by sokra — CHANGES_REQUESTED** [review]

**Comment by alexander-akait** [comment]
@xtuc ci still broken(

**Comment by sokra** [comment]
Thanks

**Pull request merged**